pkg/apis/core/v1beta1/defaults_shoot.go

```diff
@@ -0,0 +1,386 @@
+			kubeReservedMemory = resource.MustParse("1Gi")
+			kubeReservedCPU    = resource.MustParse("80m")
+			kubeReservedPID    = resource.MustParse("20k")
+		)
+
+		if obj.Spec.Kubernetes.Kubelet.KubeReserved == nil {
+			obj.Spec.Kubernetes.Kubelet.KubeReserved = &KubeletConfigReserved{Memory: &kubeReservedMemory, CPU: &kubeReservedCPU}
+			obj.Spec.Kubernetes.Kubelet.KubeReserved.PID = &kubeReservedPID
+		} else {
+			if obj.Spec.Kubernetes.Kubelet.KubeReserved.Memory == nil {
+				obj.Spec.Kubernetes.Kubelet.KubeReserved.Memory = &kubeReservedMemory
+			}
+			if obj.Spec.Kubernetes.Kubelet.KubeReserved.CPU == nil {
+				obj.Spec.Kubernetes.Kubelet.KubeReserved.CPU = &kubeReservedCPU
+			}
+			if obj.Spec.Kubernetes.Kubelet.KubeReserved.PID == nil {
+				obj.Spec.Kubernetes.Kubelet.KubeReserved.PID = &kubeReservedPID
+			}
+		}
+
+		if obj.Spec.Maintenance.AutoUpdate.MachineImageVersion == nil {
+			obj.Spec.Maintenance.AutoUpdate.MachineImageVersion = pointer.Bool(true)
+		}
+
+		if obj.Spec.Provider.WorkersSettings == nil {
+			obj.Spec.Provider.WorkersSettings = &WorkersSettings{}
+		}
+		if obj.Spec.Provider.WorkersSettings.SSHAccess == nil {
+			obj.Spec.Provider.WorkersSettings.SSHAccess = &SSHAccess{Enabled: true}
+		}
+
+		if obj.Spec.SystemComponents == nil {
+			obj.Spec.SystemComponents = &SystemComponents{}
+		}
+		if obj.Spec.SystemComponents.CoreDNS == nil {
+			obj.Spec.SystemComponents.CoreDNS = &CoreDNS{}
+		}
+		if obj.Spec.SystemComponents.CoreDNS.Autoscaling == nil {
+			obj.Spec.SystemComponents.CoreDNS.Autoscaling = &CoreDNSAutoscaling{}
+		}
+		if obj.Spec.SystemComponents.CoreDNS.Autoscaling.Mode != CoreDNSAutoscalingModeHorizontal && obj.Spec.SystemComponents.CoreDNS.Autoscaling.Mode != CoreDNSAutoscalingModeClusterProportional {
+			obj.Spec.SystemComponents.CoreDNS.Autoscaling.Mode = CoreDNSAutoscalingModeHorizontal
+		}
+	}
+
+	if obj.Spec.SchedulerName == nil {
+		obj.Spec.SchedulerName = pointer.String(v1beta1constants.DefaultSchedulerName)
+	}
+}
+
+// SetDefaults_KubeAPIServerConfig sets default values for KubeAPIServerConfig objects.
+func SetDefaults_KubeAPIServerConfig(obj *KubeAPIServerConfig) {
+	if obj.Requests == nil {
+		obj.Requests = &APIServerRequests{}
+	}
+	if obj.Requests.MaxNonMutatingInflight == nil {
+		obj.Requests.MaxNonMutatingInflight = pointer.Int32(400)
+	}
+	if obj.Requests.MaxMutatingInflight == nil {
+		obj.Requests.MaxMutatingInflight = pointer.Int32(200)
+	}
+	if obj.EnableAnonymousAuthentication == nil {
+		obj.EnableAnonymousAuthentication = pointer.Bool(false)
+	}
+	if obj.EventTTL == nil {
+		obj.EventTTL = &metav1.Duration{Duration: time.Hour}
+	}
+	if obj.Logging == nil {
+		obj.Logging = &APIServerLogging{}
+	}
+	if obj.Logging.Verbosity == nil {
+		obj.Logging.Verbosity = pointer.Int32(2)
+	}
+}
+
+// SetDefaults_KubeControllerManagerConfig sets default values for KubeControllerManagerConfig objects.
+func SetDefaults_KubeControllerManagerConfig(_ *KubeControllerManagerConfig) {}
```

Comment:
This function can be dropped.

Reply:
This function is used in one place. What if in the future we add some defaulting here and forget to call the function in that place? Maybe that's the reason why it's kept today.